website/source/docs/providers/aws/r/elasticache_subnet.html.markdown

```diff
@@ -0,0 +1,56 @@
+---
+layout: "aws"
+page_title: "AWS: aws_subnet"
+sidebar_current: "docs-aws-resource-elasticache-subnet-group"
+description: |-
+  Provides an VPC subnet resource.
+---
+
+# aws\_elasticache\_subnet\_group
+
+Provides an ElastiCache Subnet Group resource. Use this resource only when you are 
+creating an ElastiCache cluster in a VPC.
+
+## Example Usage
+
+```
+resource "aws_vpc" "foo" {
+    cidr_block = "10.0.0.0/16"
+    tags {
+            Name = "tf-test"
+    }
+}
+
+resource "aws_subnet" "foo" {
+    vpc_id = "${aws_vpc.foo.id}"
+    cidr_block = "10.0.0.0/24"
+    availability_zone = "us-west-2a"
+    tags {
+            Name = "tf-test"
+    }
+}
+
+resource "aws_elasticache_subnet_group" "bar" {
+    name = "tf-test-cache-subnet"
+    description = "tf-test-cache-subnet-group-descr"
+    subnet_ids = ["${aws_subnet.foo.id}"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `description` – (Required) Description for the cache subnet group
+* `name` – (Required) Name for the cache subnet group. This value is stored as 
```

Comment:
Random TODO: We should make this optional and use `resource.UniqueId()` if it's not specified. Same with the cache SGs